### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by xiangjunn**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -255,31 +255,58 @@ _{Explain here how the data archiving feature will be implemented}_
...
-* is reasonably comfortable using CLI apps
-
-**Value proposition**: manage contacts faster than a typical mouse/GUI driven app
+* Need to manage large number of contacts and events
+* Prefer desktop apps over other types
+* Can type fast and prefer typing to mouse interactions
+* Are reasonably comfortable using CLI apps
 
+**Value proposition**: Manage contacts of peers, professors and events such as classes and Co-Curricular Activities in
+a single system faster than a typical mouse/GUI driven app while maintaining user-friendliness
```
Comment (janjanchen) [inline docs/DeveloperGuide.md:266]: should we have space after "/"?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -255,31 +255,58 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `* * *`  | user                                       | add a new person               |                                                                        |
-| `* * *`  | user                                       | delete a person                | remove entries that I no longer need                                   |
-| `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
-| `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
-| `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
-
-*{More to be added}*
+| `* * *` | hardworking student | add contact of my TA/prof (phone number, email address, tele handle, zoom link/class room location) | save their contacts and ask them questions on my modules. |
```
Comment (janjanchen) [inline docs/DeveloperGuide.md:276]: 'Prof' instead of 'prof'? 
'telegram handle' instead of 'tele handle'?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -255,31 +255,58 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* *` | user with many contacts | add the profile picture of TA/Prof | more easily differentiate my contacts and remember their faces |
+| `*` | first time user | import existing contacts from my phone | easily access all my past contacts using SoConnect |
+| `*` | long term user | archive some contacts | still save old contacts without cluttering my current screen |
+
+#### Events
+
+| Priority | As a …​                                    | I want to …​                     | So that I can…​                                             |
+| -------- | ------------------------------------------ | ------------------------------ | ---------------------------------------------------------------------- |
+| `* * *` | SoC student | add details of event | keep track of the CCA dates and time |
+| `* * *` | SoC student | to delete details of event | remove events that have finished or are cancelled |
```
Comment (janjanchen) [inline docs/DeveloperGuide.md:293]: to remove 'to' for line 293 and 294

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (janjanchen) [reply docs/DeveloperGuide.md:276]: Maybe can mention the long form of TA and prof before we short form it?

**Review by janjanchen — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (xiangjunn) [reply docs/DeveloperGuide.md:266]: The original develop guide does not leave space after "/", so should we just stick to this format? Or do you think having a space after "/" is better? What do you think @Gordon25?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -255,31 +255,58 @@ _{Explain here how the data archiving feature will be implemented}_
...
-
-*{More to be added}*
+| `* * *` | hardworking student | add contact of my Teaching Assistants (TA)/Prof (phone number, email address, telegram handle, zoom link/class room location) | save their contacts and ask them questions on my modules. |
+| `* * *` | careless student | edit the contact of my TA/Prof | correct any mistakes or change the details when my TA changes number |
+| `* * *` | senior SoC student | delete the contact of my TA/Prof | remove contact of my TA after I have completed the module |
+| `* * *` | SoC student | view the contact of my TA/Prof | contact them when necessary |
+| `* * *` | year 4 SoC student with many contacts | search for contact of my TA/Prof | contact them when necessary |
+| `* *` | year 4 SoC student with many contacts | sort the contacts of my TA | view the contacts based on the sorting settings |
+| `* *` | careless student | undo my last action(s) | recover contacts I accidentally deleted/changed |
+| `* *` | organised SoC students | categorize the contacts of students/TA//Prof | view them separately |
```
Comment (janjanchen) [inline docs/DeveloperGuide.md:283]: there is an extra "/" before Prof!

**Review by janjanchen — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -255,57 +255,84 @@ _{Explain here how the data archiving feature will be implemented}_
...
+* 1a. SAS detects an error in the inputs.
   
-    * 1a1. SAS outputs the error message to the user.
+    * 1a1. SAS requests for correct inputs.
       
-    * 1a2. User types in a new command based on steps 1 to 2.
+    * 1a2. User enters new inputs.
       
-    * 1a3. Repeat steps 1a to 1a2 until the command entered is of correct syntax.
+    * 1a3. Repeat steps 1a1 to 1a2 until the inputs entered are correct.
```
Comment (janjanchen) [inline docs/DeveloperGuide.md:335]: Follow the format of the other use cases?

**Review by janjanchen — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -255,66 +255,93 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `* *` | SoC student with many different events to manage | categorize my events according to different tags like classes and CCAs | easily filter and search for the events I want |
+| `*` | SoC student who uses other calendars | import and export my events to other calendars like Google Calendar | easily switch calendars to manage my events |
+| `*` | long term user | archive some events that have ended | still save details of past events without cluttering my current screen |
+| `*` | Soc student with many commitments | have a reminder for upcoming events | remember to attend those events |
+
+#### Personalisation and Others
+
+| Priority | As a …​                                    | I want to …​                     | So that I can…​                                             |
+| -------- | ------------------------------------------ | ------------------------------ | ---------------------------------------------------------------------- |
+| `* * *` | first time user | have a help message | know which are the basic features, which controls there are and how to start using |
```
Comment (Gordon25) [inline docs/DeveloperGuide.md:308]: add "i can refer to" after help message
Change last column to "understand and start using basic features"

**Review by Gordon25 — CHANGES_REQUESTED** [review]
Overall, no formatting issues 👍 
For changes, you can refer to the DG notes in our google drive.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -255,66 +255,93 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
-| `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
-| `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
-
-*{More to be added}*
+| `* * *` | hardworking student | add contact of my Teaching Assistants (TA)/Prof (phone number, email address, telegram handle, zoom link/class room location) | save their contacts and ask them questions on my modules. |
+| `* * *` | careless student | edit the contact of my TA/Prof | correct any mistakes or change the details when my TA changes number |
+| `* * *` | senior SoC student | delete the contact of my TA/Prof | remove contact of my TA after I have completed the module |
+| `* * *` | SoC student | view the contact of my TA/Prof | contact them when necessary |
```
[the remark this answers is not in the document]
Comment (xiangjunn) [reply docs/DeveloperGuide.md:279]: if we remove this line, we might have no user story to link to our implementation of `list`?

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -260,10 +260,10 @@ _{Explain here how the data archiving feature will be implemented}_
 * Need to manage large number of contacts and events
 * Prefer desktop apps over other types
 * Can type fast and prefer typing to mouse interactions
-* Are reasonably comfortable using CLI apps
+* Are reasonably comfortable using *CLI* apps
 
-**Value proposition**: Manage contacts of peers, professors (Prof) and events such as classes and Co-Curricular Activities in
-a single system faster than a typical mouse/ GUI driven app while maintaining user-friendliness
+**Value proposition**: Manage contacts of peers, *Prof* and events such as classes and Co-Curricular Activities in
```
Comment (Gordon25) [inline docs/DeveloperGuide.md:265]: *Profs*, add a "s"

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -291,21 +291,21 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-| `* * *` | SoC student | search for an event based on event name | know more about the event |
-| `* *` | SoC student | sort the events by time | know which event is coming up next |
-| `* *` | SoC student with busy schedule | make sure my new event does not clash with any of the current event | better plan my timetable for additional activities |
-| `* *` | SoC student with many different events to manage | categorize my events according to different tags like classes and CCAs | easily filter and search for the events I want |
+| `* * *` | SoC student | search for an event based on event name | easily refer to the event details |
+| `* *` | SoC student | sort the events by time | prepare for upcoming events |
+| `* *` | SoC student with busy schedule | check if the new event clashes with any of my current events | better plan my timetable and avoid event clashes |
+| `* *` | SoC student with many different events to manage | categorize my events with different tags like classes and CCAs | search related events |
 | `*` | SoC student who uses other calendars | import and export my events to other calendars like Google Calendar | easily switch calendars to manage my events |
```
Comment (Gordon25) [inline docs/DeveloperGuide.md:300]: the last column should be "Synchronize my events across my calendars"

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -461,38 +461,38 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-**Guarantees:** The displayed list of events is sorted lexicographically according to the specified field, if the given field and command syntax are valid.
+**Guarantees:** The displayed list of events is sorted lexicographically according to the specified field, if the given field is valid.
 
-**MSS**
+***MSS***
 
 1. User decides on a field to sort by.
 
 2. User sorts the list.
```
Comment (Gordon25) [inline docs/DeveloperGuide.md:470]: Can change to User inputs the specific field.

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -532,7 +532,13 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 * Telegram handle: Username of a [Telegram](https://telegram.org/) user.
 
 
-* Contact details: Name and email of the contact. Phone number, address, telegram handle, video conferencing meeting link, and tag(s) are optional.
+* Prof: A group of professors who are facilitators of a specified module.
```
Comment (Gordon25) [inline docs/DeveloperGuide.md:535]: Change to Professor (Prof): a lecturer who teaches a module in NUS.

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -532,7 +532,13 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+
+* TA: A group of teaching assistants who are facilitators of a specified module.
```
Comment (Gordon25) [inline docs/DeveloperGuide.md:538]: Change to Teaching Assistant (TA): A student tutor employed to teach small group lessons in NUS.

**Review by Gordon25 — CHANGES_REQUESTED** [review]
Great job Updating the changes! 👋   Just a few nits

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Gordon25) [reply docs/DeveloperGuide.md]: I think we can follow the original developer guide.

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -255,68 +255,95 @@ _{Explain here how the data archiving feature will be implemented}_
...
-| `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
-| `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
-| `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
-
-*{More to be added}*
+| `* * *` | hardworking student | add contact of my *TA*/*Profs* | save their contacts and ask them questions on my modules. |
+| `* * *` | careless student | edit the contact of my *TA*/*Profs* | correct mistakes while adding contacts or update my TA contact details |
+| `* * *` | senior SoC student | delete the contact of my *TA*/*Profs* | remove contact of my *TA* after I have completed the module |
+| `* * *` | SoC student | view the contact of my *TA*/*Profs* | have a rough overview of my existing contacts |
```
Comment (Gordon25) [inline docs/DeveloperGuide.md:279]: Should we remove the last column?

**Review by Gordon25 — CHANGES_REQUESTED** [review]
Just 1 minor change, thats all

**Review by Gordon25 — APPROVED** [review]
LGTM!

**Pull request merged**